packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -772,9 +851,12 @@ export default class GoogleCalendarService implements Calendar {
     }
     const getCalIds = async () => {
       if (selectedCalendarIds.length !== 0) return selectedCalendarIds;
-      const cals = await this.getAllCalendars(calendar, ["id"]);
+      const cals = await this.getAllCalendars(calendar, ["id", "primary"]);
       if (!cals.length) return [];
-      return cals.reduce((c, cal) => (cal.id ? [...c, cal.id] : c), [] as string[]);
+      if (!fallbackToPrimary) return this.getValidCalendars(cals).map((cal) => cal.id);
```

Comment:
Default behaviour where we fallback to all the calendars being used for conflict checking. Remains same for regular credential.

But remember that in _getCalendarsEvents_ we don't call the getAvailability if there are no selectedCalendars, https://github.com/calcom/cal.com/blob/calendar-cache-dwd-support/packages/lib/getCalendarsEvents.ts#L52. So this code seems to be unused at the moment.